### ISSUE TITLE
Keeps Reconnecting, when changing internet connection it said ECONNRESET

### ISSUE THREAD

**Issue by KevinOktoaria** [emqx/MQTTX#1552]
#### What did I do

I change my front-end code of my website then I pushed it to my GitHub.

#### What happened

It keeps reconnecting and when I use different internet connection (my cellphone Hotspot), I got an error says ECONNRESET.

#### Environment

- OS: Windows 10
- MQTTX version, e.g. v1.9.8

#### Log detail

[2024-01-09 22:42:31] [INFO] Cuaca_TA reconnect: 0 times retry
[2024-01-09 22:42:41] [INFO] Cuaca_TA connect close, MQTT.js onClose trigger
[2024-01-09 22:42:46] [INFO] Cuaca_TA reconnect: 1 times retry
[2024-01-09 22:42:57] [INFO] Cuaca_TA connect close, MQTT.js onClose trigger
[2024-01-09 22:43:02] [INFO] Cuaca_TA reconnect: 2 times retry
[2024-01-09 22:43:13] [INFO] Cuaca_TA connect close, MQTT.js onClose trigger
[2024-01-09 22:43:17] [WARN] MQTTX force close the connection Cuaca_TA (clientID mqttx_0bd14c5a), reset the reconnect times to 0

**Comment by ysfscream** [comment]
Hi @KevinOktoaria ,

Firstly, apologies for the delayed response. I've reviewed the issue you reported regarding the persistent reconnecting behavior and the ECONNRESET error when switching to a different internet connection, such as your cellphone hotspot.

Although I do not entirely understand your issue, I've adjusted the logic for stopping reconnections in the latest update #1560 . I don't know if this will resolve your problem, but it's a step toward addressing the erratic reconnect behavior you've experienced.
